### ISSUE TITLE
[Ruby] Highlight `#{}` interpolation in heredoc

### ISSUE THREAD

**Issue by zufuliu** [ScintillaOrg/lexilla#134]
Per https://docs.ruby-lang.org/en/master/syntax/literals_rdoc.html#label-Here+Document+Literals
interpolation is allowed in double (or none) quoted or backtick quoted heredoc (`SCE_RB_HERE_QQ` and `SCE_RB_HERE_QX`).

```ruby
# SCE_RB_HERE_Q
puts <<-'EXPECTED'
One plus one is #{1 + 1}
EXPECTED

# SCE_RB_HERE_QQ
puts <<-EXPECTED.chomp
One plus one is #{1 + 1}
EXPECTED

# SCE_RB_HERE_QX
puts <<-`HEREDOC`
cat #{__FILE__}
HEREDOC
```

**Comment by zufuliu** [comment]
[ruby-0215.zip](https://github.com/ScintillaOrg/lexilla/files/10743043/ruby-0215.zip)

Patch contains three changes:
1. Set `SCE_RB_HERE_QQ` and `SCE_RB_HERE_QX` based on `HereDoc.Quoted`.
2. Add `InnerExpression` class to encapsulate `enterInnerExpression()` and `exitInnerExpression()`.
3. Highlight `#{}` in `SCE_RB_HERE_QQ` and `SCE_RB_HERE_QX`.

**Comment by nyamatongwe** [comment]
Included in 5.2.3.
